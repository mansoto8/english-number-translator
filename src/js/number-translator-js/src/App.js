import React, {Component} from 'react';
import './App.css';
import TranslatorForm from "./components/translatorForm/TranslatorForm";
import TranslationHistory from "./components/translationHistory/TranslationHistory";

class App extends Component {
    async componentDidMount() {
        this.setState(this.state);
    }

    render() {
        return (
            <div className="App">
                <header className="App-header">
                    English Number Translator
                </header>
                <TranslatorForm/>
                <TranslationHistory/>
            </div>
        );
    }
}

export default App;
