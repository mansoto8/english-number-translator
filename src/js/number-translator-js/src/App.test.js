import App from './App';
import React from 'react'
import renderer from 'react-test-renderer';

it('renders App', () => {
    const tree = renderer.create(
        <App/>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
