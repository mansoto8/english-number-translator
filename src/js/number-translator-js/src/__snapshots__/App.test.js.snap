// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    English Number Translator
  </header>
  <div>
    <div
      className="container"
    >
      <form
        className="translator-form"
        onSubmit={[Function]}
      >
        <div
          className="input-group"
          id="inputGroupNumber"
        >
          <input
            className="is-valid form-control"
            id="number-input"
            name="number"
            onChange={[Function]}
            placeholder="Enter a number"
            type="text"
            value=""
          />
          <div
            className="invalid-feedback"
            id="formFeedBackNumber"
            invalid="true"
          >
            
          </div>
        </div>
        <div
          className="form-group"
        >
          <button
            aria-label={null}
            className="btn btn-primary"
            id="translate-button"
            onClick={[Function]}
            type="submit"
          >
            Translate
          </button>
        </div>
      </form>
      <div
        className="result-div"
      >
        <label
          className=""
          htmlFor="translation-output"
          id="translation-output-label"
        >
          The number translated is: 
        </label>
        <input
          className="form-control"
          id="translation-output"
          name="translation-output"
          readOnly={true}
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <div>
    <div
      className="container"
    >
      <button
        aria-label={null}
        className="btn btn-primary"
        id="history-button"
        onClick={[Function]}
        type="button"
      >
        Check Translation History
      </button>
      <table
        id="history-table"
      >
        <thead>
          <tr>
            <th>
              Number
            </th>
            <th>
              Translation
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
      <p>
        No registers
      </p>
    </div>
  </div>
</div>
`;
